common/download.go

```diff
@@ -290,6 +290,8 @@ func (d *HTTPDownloader) Download(dst *os.File, src *url.URL) error {
 				}
 			}
 		}
+	} else if err != nil || (resp.StatusCode >= 400 && resp.StatusCode < 600) {
+		return fmt.Errorf("%s", resp.Status)
```

Comment:
I'd like a slightly more descriptive error message explaining when this is failing.